### ISSUE TITLE
New camera Hualai

### ISSUE THREAD

**Issue by dmolner** [EliasKotlyar/Xiaomi-Dafang-Hacks#541]
Hello
Do you think is possible include this cam in this project?
[Xiaomi Hualai Xiaofang 1080P Dual Lens Panoramic View Smart WIFI IP Camera Xiaomi Smart Home
https://banggood.app.link/NDhvKEcsdO]url)

**Comment by jmtatsch** [comment]
Hard to say at this time. It may be using dual T20s.
You should order one and tear it down.

**Comment by EliasKotlyar** [comment]
I dont think that there is a T20 Processor inside, because according to the specs provided with the sdk, it cannot handle more than 1 sensor. But we will see as soon as someone gets one and tear it down.

**Comment by dmolner** [comment]
Ok,
Is possible a T20 processor because the pub say two processors.

**Comment by maxhammond** [comment]
Just got one of these cameras and the 360 viewing is really cool. Really worth taking a look at.

**Comment by EliasKotlyar** [comment]
Can you try to disassemble it and check which processor is inside?

**Comment by gozhev** [comment]
Here is a teardown https://www.zhihu.com/question/277660075/answer/395470236
Yet another new SOC, guys.

**Comment by saneklad** [comment]
Any news for supporting this device?

**Comment by craigdent** [comment]
Is it possible to hack this camera?

**Comment by craigdent** [comment]
Do you think you’ll make a hack for this camera anytime soon?

Would love to incorporate it into homebridge/HomeKit

**Comment by EliasKotlyar** [comment]
Hello Craigdent,

I suppose that it wont happen so fast. Its almost every time a really rocky path to create such a hack. First requirement is, that you have an SDK for the required SoC. Without an SDK, its nearly impossible to make all the functionality like the RTSP-Server work. In this case, the SoC is named "M3C SiP":
http://www.vatics.com/M3C_SiP_Data_Brief.pdf

Unfortunately, i cannot find a SDK for this SDK trough google. So first, we need to wait until someone provides this SDK. You can ask the manufacturer of this SoC, but they rarely provide an SDK for free. 

So for now, i am closing this topic, because we dont have any possibility to hack it yet.

**Comment by Moskvitch** [comment]
Just received my "two-faced Janus" Xiaofang Hualai.  As it came with a loose part making a rattling noise :) I had a perfect excuse to take it a part to avail myself of the spare screw (as it turned out) provided by Xiaomi :) :) :)
So, for what it's worth, inside the biggest CPU-looking chip is a
VATICS
M3881C
1725
7SQP10284.01.1

**Comment by craigdent** [comment]
anyone found a way to enable the RTSP or hack this camera?

**Comment by minzukas** [comment]
Following

**Comment by kbesson** [comment]
Up ?

Got mine today and disassembled it. I spoted a 3 pins unpopulated connector. I will try to hookup a serial connection to see what's inside.

We can find this generic board on AliExpress for 40 bucks.

**Comment by minzukas** [comment]
Following

**Comment by loglux** [comment]
No progress?
